universe/ValueRefs.cpp

```diff
@@ -310,20 +306,24 @@ std::string ValueRefBase::InvariancePattern() const {
         .append(ConstantExpr()                  ? "C" : "c");
 }
 
-MeterType NameToMeter(const std::string& name) {
-    MeterType retval = MeterType::INVALID_METER_TYPE;
-    auto it = GetMeterNameMap().find(name);
-    if (it != GetMeterNameMap().end())
-        retval = it->second;
-    return retval;
+MeterType NameToMeter(const std::string_view name) {
+    auto it = std::find_if(NAME_TO_METER.begin(), NAME_TO_METER.end(),
+        [&name](const std::pair<std::string_view, MeterType>& elem) {
+            return elem.first == name;
+        }
+    );
+
+    return (it != NAME_TO_METER.end()) ? it->second : MeterType::INVALID_METER_TYPE;
 }
 
-const std::string& MeterToName(MeterType meter) {
-    for (auto& [name, type] : GetMeterNameMap()) {
-        if (type == meter)
-            return name; // CodeQL reports "Returning stack-allocated memory", but the name structured binding should be a reference as far as I can tell
-    }
-    return EMPTY_STRING;
+const std::string& MeterToName(const MeterType& meter) {
+    auto it = std::find_if(NAME_TO_METER.begin(), NAME_TO_METER.end(),
+        [&meter](const std::pair<std::string, MeterType>& elem) {
+            return elem.second == meter;
+        }
+    );
+
+    return (it != NAME_TO_METER.end()) ? it->first : EMPTY_STRING;
 }
 
 constexpr std::string_view PlanetTypeToStringConstexpr(PlanetType type) {
```

Comment:
I can later have a look if it's possible to simplify the PlanetType code here to also use an array of pairs.